recipes/xcube/meta.yaml

```diff
@@ -0,0 +1,87 @@
+{% set name = "xcube" %}
+{% set version = "0.2.0.dev1" %}
+
+package:
+  name: "{{ name }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://github.com/dcs4cop/{{ name }}/archive/{{ version }}.tar.gz"
+  sha256: "2617a04fb38511b9bba629e57c5358c4c2cb2384ec216334dd6f6947b43a80a3"
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+  entry_points:
+    # Put any entry points (scripts to be generated automatically) here.
+    - xcube = xcube.cli.cli:main
+    - xcube-gen = xcube.cli.gen:main
+    - xcube-restime = xcube.cli.restime:main
+
+requirements:
+  build:
+    # Python
+    - python =3.7
+    - pip =19.2
+  run:
+    # Python
+    - python =3.7
+    # Required
+    - affine =2.2
```

Comment:
These pinning look all much too strict. Why is this so?

Reply:
Can you clarify whether you are referring to pythin and affine or to all version pinnings? 

Thanks for all the comments BTW. Very useful indeed.

Reply:
All version pinnings. They are so specific that you will quite often run into situations where `xcube` cannot be installed alongside other packages in a conda environment.

Reply:
> These pinning look all much too strict. Why is this so?

We'd love to relax the version requirements!

It is so strict because we very often run into dependency hell mainly caused by gdal and its shared libraries and all its transitive dependencies in combination with other packages such as xarray and matplotlib.

Reply:
But a lot of packages like `pip`, `pyyaml`, `s3fs`, … should be impacted by that. Please loosen that pinning. Otherwise we have https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml for global conda-forge pinning that should avoid dependency hell. We have a bot that automatically updates packages when the global pinning changes. This should avoid dependency hell with gdal and matplotlib. These global pinnings only work when you don't have an explicit pinning in the recipe (`conda smithy` takes care that the global pinning are applied here). Otherwise, exact pinning in a single package is normally the way into dependency hell.

Reply:
Changed version pinnings to >=. please advice whether this suffices